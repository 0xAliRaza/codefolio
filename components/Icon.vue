<template>
  <svg-icon :name="name" :title="title" :desc="desc"></svg-icon>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({
  props: {
    name: { type: String, required: true },
    title: { type: String, default: null },
    desc: { type: String, default: null },
  },
})
</script>
<style lang="scss" scoped>
.icon {
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: 1em;
  height: 1em;
  stroke-width: 0;
  stroke: currentColor;
  fill: currentColor;
}
</style>
