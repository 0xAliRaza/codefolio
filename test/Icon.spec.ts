import { shallowMount } from '@vue/test-utils'
import Icon from '@/components/Icon.vue'

const DUMMY_PROPS = {
  title: 'Test title',
  desc: 'Test Desc',
  name: 'test_name',
}
const factory = () => {
  return shallowMount(Icon, {
    propsData: DUMMY_PROPS,
    stubs: ['svg-icon'],
  })
}

describe('Icon', () => {
  it('should render properly', () => {
    const wrapper = factory()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should should contain props', () => {
    const wrapper = factory()
    const attrs = wrapper.attributes()
    expect(attrs.title).toMatch(DUMMY_PROPS.title)
    expect(attrs.desc).toMatch(DUMMY_PROPS.desc)
    expect(attrs.name).toMatch(DUMMY_PROPS.name)
  })
})
