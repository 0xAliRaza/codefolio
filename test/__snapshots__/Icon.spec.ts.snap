// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render properly 1`] = `
<svg-icon-stub name="test_name"
               title="Test title"
               desc="Test Desc"
>
</svg-icon-stub>
`;
